resources/assets/lib/beatmap-discussions/review-post.tsx

```diff
@@ -37,16 +44,44 @@ export class ReviewPost extends React.Component<Props> {
             'strong',
             'text',
           ]}
-          plugins={[
-            MarkdownEmbedTokenizer,
-          ]}
-          source={this.props.message}
+          key={osu.uuid()}
+          source={source}
           renderers={{
-            embed: ReviewPostEmbed,
             link: (props) => <a className='beatmap-discussion-review-post__link' rel='nofollow' {...props}/>,
-            paragraph: (props) => <div className='beatmap-discussion-review-post__block' {...props}/>,
+            paragraph: (props) => {
+              return <div className='beatmap-discussion-review-post__block'>
+                <div className='beatmapset-discussion-message' {...props}/>
+              </div>;
+            },
           }}
         />
+    );
+  }
+
+  render() {
+    const docBlocks: JSX.Element[] = [];
+
+    try {
+      const document = JSON.parse(this.props.message);
+
+      _.each(document, (block) => {
+        switch (block.type) {
+          case 'paragraph':
+            // '&nbsp;  ' converts into a newline
+            docBlocks.push(this.paragraph(osu.presence(block.text.toString()) || '&nbsp;  '));
```

Comment:
On `.toString()`, I'm thinking more handling it server side so the type will be consistent. Also you can currently store weird stuff in text field (something like `{type:"paragraph",text:[[1]]}`)